### PR TITLE
JSHintBear.py: Update jshint to 2.9.5

### PR DESCRIPTION
**Pull request by margobra8**
This updates the npm jshint requirement to
its most recent version and accommodates
new changes.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by jayvdb** [comment]
You need to update also `package.json` , which is kept in sync using `.ci/generate_bear_requirements.py`

**Comment by margobra8** [comment]
(sorry for closing, accidentally rebased the wrong HEAD)

**Comment by jayvdb** [comment]
No worries.  Glad to see you recovered your PR ;-)

It is currently needing a rebase.

Also the commit message needs to comply with http://coala.io/commit 

I've created https://github.com/coala/coala-bears/issues/2201 for this.

**Comment by margobra8** [comment]
What do you mean now with `rebase`? A rebase I need to perform?

File: bears/js/JSHintBear.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class JSHintBear:
     """
 
     LANGUAGES = {'JavaScript'}
-    REQUIREMENTS = {NpmRequirement('jshint', '2')}
+    REQUIREMENTS = {NpmRequirement('jshint', '2.9.5')}
```
Comment (Monal5031) [inline bears/js/JSHintBear.py]: Shouldn't we update it in `package.json` too?

Comment (margobra8) [reply bears/js/JSHintBear.py]: It is

Comment (Monal5031) [reply bears/js/JSHintBear.py]: yeah my bad, didn't scroll fully. Sorry for the inconvenience

**Comment by Mixih** [comment]
@gitmate rebase

**Comment by margobra8** [comment]
@Mixih how come I do that?

**Comment by Monal5031** [comment]
@margobra8 You'll have to do it locally, refer [http://api.coala.io/en/latest/Developers/Git_Basics.html#rebasing](http://api.coala.io/en/latest/Developers/Git_Basics.html#rebasing)

**Comment by margobra8** [comment]
Rebased

**Comment by Mixih** [comment]
ack ea61fef

**Comment by Makman2** [comment]
Actually I dislike the commit message: The goal was to add a new setting, not to upgrade jshint. This was just a consequence. Please adapt your message.
Also: Please don't put a colon between `Closes` and the issue id.

**Comment by Makman2** [comment]
unack ea61fef

**Comment by margobra8** [comment]
@Makman2 the gci task was about updating the jshint version and implementing any changes that needed to be added as a consequence of the upgrade. So it’s the other way round.

**Comment by Makman2** [comment]
Could you provide me the link to the GCI task? :)
(Sorry I'm reviewing many PRs independent of GCI, and I lost a bit track what is what :/)

However, the issue reads
> Add JSHint option enforce_trailing_comma

Updating the jshint version doesn't require adding settings, that's an additional feature.

**Comment by margobra8** [comment]
But then I don’t understand why @jayvdb opened that issue

**Comment by Makman2** [comment]
> Updating the jshint version doesn't require adding settings, that's an additional feature.

What I want to say that it's usually the other way around ;)
The goal is to add a new analysis feature to the bear. This seems to require updating jshint because it introduces this analysis feature in this new versions as far as I understand.

> Does that issue need to be closed?

Don't think so ;) I think if you change the commit message to reverse the reason then everything should be fine :)

I hope I could clear your doubts :) If not just ask :+1:

**Comment by margobra8** [comment]
Ok maybe it’s best to change the commit message

File: bears/js/JSHintBear.py (hunk 1 above)
Comment (newbazz) [reply bears/js/JSHintBear.py]: Unnecessary change ;P

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "eslint-plugin-import": "~2",
     "happiness": "~10",
     "htmlhint": "~0.9.13",
-    "jshint": "~2",
+    "jshint": "~2.9.5",
```
Comment (newbazz) [inline package.json]: This is also unecessary

**Comment by margobra8** [comment]
@newbazz please read above, in order to add this option it’s necessary to update the jshint requirement.

**Comment by margobra8** [comment]
It was also the original purpose of this pull request but I changed the commit message as @Makman2 suggested.

**Comment by Makman2** [comment]
@newbazz was heedful and checked it for me again if the version change is necessary, and he told me that the current version also offers the feature you try to implement. Meaning an upgrade is not necessary in this commit :) I hope we don'T cause you too much trouble, but we try to make commits as atomic as possible. Ideally one should be able to understand coala completely by reading each commit (and its attached issue reference).

**Comment by Makman2** [comment]
Just to be sure @newbazz : Have you tested against the minimum version we specify, so in this case v2.0.0 ?

**Comment by margobra8** [comment]
According to [jshint 2.9.5 release](https://github.com/jshint/jshint/releases/tag/2.9.5) it was in this version when the feature was added.

**Comment by newbazz** [comment]
@Makman2 i read it in one of the stack answers

**Comment by newbazz** [comment]
@Makman2 but in the release list of the jshint repo it is mentioned
image
But then we have created a new issue for it right.

**Comment by Makman2** [comment]
Then the update is correct.

**Comment by Makman2** [comment]
Alright, please just remove the colon in the issue reference: `Closes: #2201` --> `Closes ...`

After that it's good for merge ;)

**Comment by Makman2** [comment]
unack 0c4641fd087dd002bd89bc375e010629bf315fe1

**Comment by margobra8** [comment]
Done

**Comment by Makman2** [comment]
ack c8250e6

**Comment by Makman2** [comment]
Merging as soon as CI is green ;)

**Comment by Makman2** [comment]
@gitmate-bot ff

**Pull request merged**